app/api/routes.py

```diff
@@ -16,9 +16,14 @@ def resources():
     return get_resources()
 
 
-@bp.route('/resources/<int:id>', methods=['GET'])
-def resource(id):
-    return get_resource(id)
+@bp.route('/resources/<int:id>', methods=['GET', 'PUT'])
+def resource(id, category=None, languages=[], name=None, url=None,
```

Comment:
So I understand that `id` is being passed into this method because it's in the route itself. What is the purpose of passing in `category`, `languages`, etc? All of that info comes from `request.args`, right? These are unnecessary in the method signature, aren't they?

Reply:
The other parameters after `id` (i.e. `category`, `languages`, etc) are default values that serve to provide a field with an appropriate "empty" value if no user input is specified for that field. I was just concerned that if I simply return `request.args` for all fields then certain fields (i.e. `languages` and `paid`) would have `None` as their default values, rather than an empty list or False boolean for their defaults (which I think are more appropriate).

Reply:
Fair enough. Wouldn't it be better to just declare them as variables inside the function rather than pass them into the function? I just think it's hard to see how these are used when they're declared like this.